sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -175,6 +175,64 @@ object GpuCast extends Arm {
       cv.stringReplaceWithBackrefs(rule.search, rule.replace)
     })
   }
+
+  def sanitizeStringToFloat(input: ColumnVector): ColumnVector = {
+
+    // This regex gets applied after the transformation to normalize use of Inf and is
+    // just strict enough to filter out known edge cases that would result in incorrect
+    // values. We further filter out invalid values using the cuDF isFloat method.
+    val VALID_FLOAT_REGEX =
+      "^" +                         // start of line
+      "[+\\-]?" +                   // optional + or - at start of string
+      "(" +
+        "(" +
+          "(" +
+            "([0-9]+)|" +           // digits, OR
+            "([0-9]*\\.[0-9]+)|" +  // decimal with optional leading and mandatory trailing, OR
+            "([0-9]+\\.[0-9]*)" +   // decimal with mandatory leading and optional trailing
+          ")" +
+          "([eE][+\\-]?[0-9]+)?" +  // exponent
+          "[fFdD]?" +               // floating-point designator
+        ")" +
+        "|Inf" +                    // Infinity
+        "|[nN][aA][nN]" +           // NaN
+      ")" +
+      "$"                           // end of line
+
+    withResource(input.lstrip()) { stripped =>
+      withResource(GpuScalar.from(null, DataTypes.StringType)) { nullString =>
+        // filter out strings containing breaking whitespace
+        val withoutWhitespace = withResource(ColumnVector.fromStrings("\r", "\n")) {
+            verticalWhitespace =>
+          withResource(stripped.contains(verticalWhitespace)) {
+            _.ifElse(nullString, stripped)
+          }
+        }
+        // replace all possible versions of infinity with Inf
+        val inf = withResource(withoutWhitespace) { _ =>
+          withoutWhitespace.stringReplaceWithBackrefs(
+            "([iI][nN][fF])([iI][nN][iI][tT][yY])?", "Inf")
```

Comment:
Does this need a capture group?  This is a case where we really want to just call `stringReplaceRegex` but apparently that doesn't exist in the cudf Java bindings even though libcudf supports it. 😞  Still, I think this becomes a cheaper regex without the capture group.  Wouldn't this be equivalent?
```suggestion
            "[iI][nN][fF])([iI][nN][iI][tT][yY]", "Inf")
```

Reply:
I committed this, then saw tests fail and realized I had misread this. There are two groups here - (Inf) and (inity) and the second grouping is optional so that we replace all variations of Inf and Infinity with the standardized version,

Reply:
Oops, I made the same mistake when reading it.  Funny how the parentheses can disappear in the middle. 😅

Reply:
Since we don't care about actually capturing the groups for backreference, I think these groups should be marked as non-capturing to hopefully use a cheaper regex kernel, i.e.: something like:
```
"(?:[iI][nN][fF])(?:[iI][nN][iI][tT][yY])?"
```